src/main.rs

```diff
@@ -307,7 +307,10 @@ fn matches_desired_prefix(hash: &[u8; SHA1_BYTE_LENGTH], prefix: &HashPrefix) ->
 
 fn create_git_object_file(search_result: &HashMatch) -> io::Result<()> {
     let compressed_object = zlib_compress(&search_result.data)?;
-    let dir_path = format!(".git/objects/{:02x}", search_result.hash[0]);
+    let git_dir_bytes = run_command("git", &["rev-parse", "--git-dir"]);
+    let mut git_dir = &String::from_utf8(git_dir_bytes).expect("git rev-parse --git-dir returned invalid utf8");
+    git_dir.truncate(git_dir.len() - 1);
```

Comment:
I know this line errors, I'm debugging it now.

Reply:
You might want to use `&mut String::from_utf8...` rather than `&String::from_utf8...` on the line above.